src/flexmock/api.py

```diff
@@ -244,19 +244,22 @@ def _create_placeholder_mock_for_proper_teardown(
     def _update_method(self, expectation: "Expectation", name: str) -> None:
         method_instance = self._create_mock_method(name)
         obj = self._object
-        if self._hasattr(obj, name) and not hasattr(expectation, "_original"):
-            expectation._update_original(name, obj)
-            method_type = type(_getattr(expectation, "_original"))
-            try:
-                # TODO(herman): this is awful, fix this properly.
-                # When a class/static method is mocked out on an *instance*
-                # we need to fetch the type from the class
-                method_type = type(_getattr(obj.__class__, name))
-            except Exception:
-                pass
-            if method_type in SPECIAL_METHODS:
-                expectation._original_function = getattr(obj, name)
-            expectation._method_type = method_type
+        if self._hasattr(obj, name):
+            if not hasattr(expectation, "_original"):
+                expectation._update_original(name, obj)
+                method_type = type(_getattr(expectation, "_original"))
+                try:
+                    # TODO(herman): this is awful, fix this properly.
+                    # When a class/static method is mocked out on an *instance*
+                    # we need to fetch the type from the class
+                    method_type = type(_getattr(obj.__class__, name))
+                except Exception:
+                    pass
+                if method_type in SPECIAL_METHODS:
+                    expectation._original_function = getattr(obj, name)
+                expectation._method_type = method_type
+            else:
```

Comment:
Same remark as on the other PR, invert the condition

Reply:
Fixed and rebased with master